pkg/reconciler/nscert/nscert.go

```diff
@@ -96,11 +97,12 @@ func (c *reconciler) reconcile(ctx context.Context, ns *corev1.Namespace) error
 	// Only create wildcard certs for the default domain
 	defaultDomain := cfg.Domain.LookupDomainForLabels(nil /* labels */)
 
-	labelSelector := kubelabels.SelectorFromSet(
-		kubelabels.Set{
-			networking.WildcardCertDomainLabelKey: defaultDomain,
-		},
-	)
+	labelSelector := kubelabels.NewSelector()
+	req, err := kubelabels.NewRequirement(networking.WildcardCertDomainLabelKey, selection.Exists, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create requirement: %v", err)
+	}
+	labelSelector = labelSelector.Add(*req)
```

Comment:
I don't think you add re-assignment.

Reply:
From what I can tell `Add` returns a new Selector so I have to assign it.

Reply:
Yeah, I guess internally it's just a slice.
🤷‍♂